src/services/BuildingService.ts

```diff
@@ -1,68 +1,66 @@
-import axios from 'axios';
+import { typedRequest } from '../../src/services/api/typedRequest';
+import type { paths, components } from '../../src/services/api/platform-schema';
 
-export interface BuildingUnit {
-  id?: string;
-  title: string;
-  description?: string;
-  location?: string;
-  livingSpace?: number;
-  commercialSpace?: number;
-  usableSpace?: number;
-  heatingSpace?: number;
-  address?: {
-    street: string
-    city: string
-    province: string
-    zip: string
-    country: string
-  }
-}
+// OpenAPI schema types
+export type CreateBuildingRequest =
+  paths['/api/v1/projects/{projectId}/properties/{propertyId}/buildings']['post']['requestBody']['content']['application/json'];
 
-class BuildingService {
-  private readonly baseUrl: string = '/api/v1/projects';
+// There is no GET success response, so we fallback to BuildingJson directly wait until backend update 
+export type BuildingResponse = components['schemas']['BuildingJson'];
 
-  async createBuilding(
-    projectId: string,
-    propertyId: string,
-    building: BuildingUnit,
-  ): Promise<BuildingUnit> {
-    return axios
-      .post(`${this.baseUrl}/${projectId}/properties/${propertyId}/buildings`, building)
-      .then((response) => {
-        console.debug(response);
-        return response.data;
-      });
+export type UpdateBuildingRequest =
```

Comment:
We should not use different types for request and response. It is in both cases a BuildingJson.

Could you please implement this as cleanly as in the ProjectService.